test/steps/v2_steps.py

```diff
@@ -58,6 +66,42 @@ def parse_bool(value):
 register_type(MaybeBool=parse_bool)
 
 
+########### GENERIC HELPERS ############
+
+
+def load_resource(res, is_binary=True):
+    """load data from features/resources"""
+    path = Path(__file__).parent.parent / "features" / "resources" / res
+    filemode = "rb" if is_binary else "r"
+    with open(path, filemode) as fin:
+        data = fin.read()
+    return data
+
+
+def read_program_binary(path):
+    return bytearray(load_resource(path))
+
+
+def read_program(context, path):
+    """
+    Assumes that have already added `context.app_acl` so need to have previously
+    called one of the steps "Given an algod v2 client..."
+    """
+    if path.endswith(".teal"):
```

Comment:
in the case of `*.teal`, compile using **algod**'s `/compile` endpoint (turned on in developer mode)